pkg/runtime/runtime.go

```diff
@@ -291,11 +301,34 @@ func (a *DaprRuntime) initRuntime(opts *runtimeOpts) error {
 	log.Infof("internal gRPC server is running on port %v", a.runtimeConfig.InternalGRPCPort)
 
 	// Start HTTP Server
-	a.startHTTPServer(a.runtimeConfig.HTTPPort, a.runtimeConfig.ProfilePort, a.runtimeConfig.AllowedOrigins, pipeline)
+	a.startHTTPServer(a.runtimeConfig.HTTPPort, a.runtimeConfig.ProfilePort, a.runtimeConfig.AllowedOrigins, httpPipeline)
 	log.Infof("http server is running on port %v", a.runtimeConfig.HTTPPort)
 
 	return nil
 }
+func (a *DaprRuntime) buildGRPCPipeline() (dapr_grpc_middleware.Pipeline, error) {
+	var interceptors []dapr_grpc_middleware.Middleware
```

Comment:
Seems len(interceptors) == len(a.globalConfig.Spec.GRPCPipelineSpec.Handlers), so may be we can preallocate here.